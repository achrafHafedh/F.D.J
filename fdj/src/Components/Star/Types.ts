export type StarProps = {
  index: number;
  value: number;
  dispatch: Function;
  className: string;
};
