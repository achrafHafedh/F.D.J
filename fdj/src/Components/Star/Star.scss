.container-star {
  position: relative;
  width: 35px;
  display: inline-block;
  margin: 0 5px;

  .star::after {
    content: "\2605";
    font-size: 2rem;
    color: white;
  }

  input:checked {
    ~ .star::after {
      content: "\2605";
      color: #fc0;
    }
    ~ label {
      color: white;
    }
  }

  label {
    position: absolute;
    top: 12px;
    left: 0px;
    color: #fc0;
    font-size: 0.8rem;
    cursor: pointer;
    padding: 4px 10px;
    font-weight: bold;
    &.big {
      left: -4px;
    }
  }

  .visually-hidden {
    position: absolute;
    left: -100vw;
  }
}
