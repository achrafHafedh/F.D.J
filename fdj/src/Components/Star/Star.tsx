import { FormEvent, FunctionComponent } from "react";
import { StarProps } from "./Types";

import "./Star.scss";

const Star: FunctionComponent<StarProps> = ({ index, value, onClick, className }) => {
  return (
    <div className="container-star">
      <input
        type="checkbox"
        onChange={(v: FormEvent<HTMLInputElement>) =>
          console.log({ v: v.currentTarget.value, checked: v.currentTarget.checked })
        }
        id={`star-${index}`}
        className="visually-hidden"
        value={value}
      />
      <label className={className} htmlFor={`star-${index}`}>
        {value}
      </label>
      <div className="star"></div>
    </div>
  );
};

export default Star;
