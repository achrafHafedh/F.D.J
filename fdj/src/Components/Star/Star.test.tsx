import { render } from "@testing-library/react";

import Star from "./Star";

describe("<Star/>", () => {
  it("should render Star with value", () => {
    const props = {
      index: 1,
      value: 1,
      dispatch: () => "",
      className: "big",
    };
    const { container } = render(<Star {...props} />);
    expect(container).toMatchSnapshot();
  });
});
