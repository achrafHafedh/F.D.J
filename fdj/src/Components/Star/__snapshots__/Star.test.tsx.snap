// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Star/> should render Star with value 1`] = `
<div>
  <div
    class="container-star"
  >
    <input
      class="visually-hidden"
      id="star-1"
      type="checkbox"
      value="1"
    />
    <label
      class="big"
      for="star-1"
    >
      1
    </label>
    <div
      class="star"
    />
  </div>
</div>
`;
